rotkehlchen/chain/ethereum/modules/balancer/balancer.py

```diff
@@ -0,0 +1,687 @@
+                addresses=existing_addresses,
+                start_ts=min_end_ts,
+                end_ts=to_timestamp,
+            )
+            address_to_swaps.update(address_new_swaps)
+            self._update_used_query_range(
+                addresses=existing_addresses,
+                prefix='balancer_trades',
+                start_ts=min_end_ts,
+                end_ts=to_timestamp,
+            )
+
+        # Insert all unique swaps to the DB
+        if address_to_swaps:
+            all_swaps = {swap for a_swaps in address_to_swaps.values() for swap in a_swaps}
+            self.database.add_amm_swaps(list(all_swaps))
+
+        return self._fetch_trades_from_db(
+            addresses=addresses,
+            from_timestamp=from_timestamp,
+            to_timestamp=to_timestamp,
+        )
+
+    def _get_address_to_swaps_graph(
+            self,
+            addresses: List[ChecksumEthAddress],
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> AddressToSwaps:
+        """Get a mapping of addresses to swaps for the given time range
+
+        May raise RemoteError
+        """
+        addresses_lower = [address.lower() for address in addresses]
+        querystr = format_query_indentation(SWAPS_QUERY.format())
+        param_types = {
+            '$limit': 'Int!',
+            '$offset': 'Int!',
+            '$addresses': '[ID!]',
+            '$start_ts': 'Int!',
+            '$end_ts': 'Int!',
+        }
+        param_values = {
+            'limit': GRAPH_QUERY_LIMIT,
+            'offset': 0,
+            'addresses': addresses_lower,
+            'start_ts': start_ts,
+            'end_ts': end_ts,
+        }
+        address_to_swaps: DDAddressToSwaps = defaultdict(list)
+        while True:
+            try:
+                result = self.graph.query(
+                    querystr=querystr,
+                    param_types=param_types,
+                    param_values=param_values,
+                )
+            except RemoteError as e:
+                self.msg_aggregator.add_error(SUBGRAPH_REMOTE_ERROR_MSG.format(error_msg=str(e)))
+                raise
+
+            try:
+                raw_swaps = result['swaps']
+            except KeyError as e:
+                log.error(
+                    'Failed to deserialize balancer swaps',
+                    error='Missing key: swaps',
+                    result=result,
+                    param_values=param_values,
+                )
+                raise RemoteError('Failed to deserialize balancer trades') from e
+
+            for raw_swap in raw_swaps:
+                try:
+                    amm_swap = deserialize_swap(raw_swap)
+                except DeserializationError as e:
+                    log.error(
+                        'Failed to deserialize a balancer swap',
+                        error=str(e),
+                        raw_swap=raw_swap,
+                        start_ts=start_ts,  # initial value
```

Comment:
Why do you need those in the log? What does the comment want to say?

Reply:
I think it is useful to have at least the original `start_ts` timestamp, because in a paginated request `param_values` may contain a different `start_ts`.